examples/server/server.cpp

```diff
@@ -1311,24 +1312,45 @@ int main(int argc, char **argv)
                         sent_token_probs_index = probs_stop_pos;
                     }
 
-                    const json data = llama.has_next_token
-                                          ? format_partial_response(llama, to_send, probs_output)
-                                          // Generation is done, send extra information.
-                                          : format_final_response(llama, to_send, llama.generated_token_probs);
+                    {
+                        // Always send partial response
+                        // so we can get the correct partial response of the last to_send in the client
+                        const json data = format_partial_response(llama, to_send, probs_output);
```

Comment:
I also made the last to_send have a partial response, so we can get the probabilities of last message correctly. (the final response included all probabilities).